### PR TITLE
ENH - Add ``LogisitcGroup`` datafit

### PR DESCRIPTION
**Pull request by Badr-MOUFAD**
This adds a ``LogisticGroup`` datafit.

- [x] Implement ``LogisitcGroup`` datafit
- [x] unittest using group logistic regression

Soon, a ``GroupProxNewton`` solver :rocket:

### CODE REVIEW

File: skglm/datafits/group.py [hunk 1]
```diff
@@ -71,3 +71,78 @@ def gradient_scalar(self, X, y, w, Xw, j):
 
     def intercept_update_step(self, y, Xw):
         return np.mean(Xw - y)
+
+
+class LogisticGroup(BaseDatafit):
```
Comment (mathurinm) [inline skglm/datafits/group.py:76]: Could it inherit from Logistic to avoid duplicating code ?

File: skglm/tests/test_group.py [hunk 2]
```diff
@@ -160,6 +161,66 @@ def test_intercept_grouplasso():
...
+    rnd = np.random.RandomState(1123)
+    X, y, _ = make_correlated_data(n_samples, n_features, random_state=rnd)
+    y = np.sign(y)
+
+    grp_indices, grp_ptr = grp_converter(1, n_features)
+    weights = np.ones(n_features)
+    alpha_max = norm(X.T @ y, ord=np.inf) / (2 * n_samples)
+    alpha = rho * alpha_max / 10.
+
+    log_group = LogisticGroup(grp_ptr=grp_ptr, grp_indices=grp_indices)
```
Comment (mathurinm) [inline skglm/tests/test_group.py:176]: log means np.log for me. `logistic` removes this ambiguity

File: skglm/tests/test_group.py [hunk 3]
```diff
@@ -160,6 +161,66 @@ def test_intercept_grouplasso():
...
+
+@pytest.mark.parametrize("n_groups, rho", [[15, 1e-1], [25, 1e-2]])
+def test_group_logreg(n_groups, rho):
+    n_samples, n_features, shuffle = 30, 60, True
+    random_state = 123
+
+    X, y, _ = make_correlated_data(n_samples, n_features, random_state=random_state)
+    y = np.sign(y)
+
+    np.random.seed(random_state)
```
Comment (mathurinm) [inline skglm/tests/test_group.py:200]: usually this is frowned upo (it resets the seed globally, that's a heavy side effect) 

use rng = sklearn.utils.check_random_state(random_state), then rng.randn

File: skglm/datafits/group.py (hunk 1 above)
Comment (Badr-MOUFAD) [reply skglm/datafits/group.py]: Agree!

**Side note**: There is a small pitfall in this case as we are using jitclasses: we can't call the constructor of the ``Logistic`` (I get a numba compilation error because of the ``super`` keyword).

File: skglm/tests/test_group.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+from turtle import numinput
```
Comment (mathurinm) [inline skglm/tests/test_group.py:1]: vscode <3

**Review by mathurinm — APPROVED** [review]
